services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/QueryEnhancedRequest.java

```diff
@@ -50,7 +53,7 @@ private QueryEnhancedRequest(Builder builder) {
         this.limit = builder.limit;
         this.consistentRead = builder.consistentRead;
         this.filterExpression = builder.filterExpression;
-        this.projectionExpression = builder.projectionExpression;
+        this.attributesToProject = builder.attributesToProject;
```

Comment:
We should ensure this list cannot be modified, especially as we're vending it in a getter. I suggest ```this.attributesToProject = Collections.unmodifiableList(builder.attributesToProject);```